src/MatrixTwitter.js

```diff
@@ -0,0 +1,1040 @@
+            log.warn('Twitter', "Authentication with existing token failed. ");
+            fs.unlink('bearer.tok', (err) => {
+              if(err) {
+                log.warn('Twitter', "Couldn't delete bearer.tok");
+              }
+              resolve(this._get_bearer_http());
+            });
+          }
+          else if (response.statusCode == 200) {
+            log.info('Twitter', "Existing token OK.");
+            resolve(token);
+          }
+          else {
+            log.error("Twitter", error);
+            reject("Unexpected response to application/rate_limit_status " +
+              "during bearer token validation. Bailing.");
+          }
+        });
+    });
+  });
+}
+
+MatrixTwitter.prototype._get_intent = function (id) {
+  return this._bridge.getIntentFromLocalpart("twitter_" + id);
+}
+
+MatrixTwitter.prototype._update_user_timeline_profile = function (profile) {
+  var ts = new Date().getTime();
+  if(profile == null) {
+    log.warn("Twitter", "Tried to preform a profile update with a null profile.");
+    return;
+  }
+  return this.storage.get_profile_by_id(profile.id).then((old)=>{
+    var update_name = true;
+    var update_avatar = true;
+    if(old != null && old.profile != null) {
+      //If either the real name (name) or the screen_name (handle) are out of date, update the screen name.
+      update_name = (old.profile.name != profile.name)
+      update_name = (old.profile.screen_name != profile.screen_name);
+      update_avatar = (old.profile.profile_image_url_https != profile.profile_image_url_https);
+    }
+
+    var intent = this._get_intent(profile.id_str);
+    if(update_name) {
+      if(profile != null && profile.name != null && profile.screen_name != null ) {
+        intent.setDisplayName(profile.name + " (@" + profile.screen_name + ")");
+      }
+      else {
+        log.warn("Twitter", "Tried to preform a user display name update with a null profile.");
+      }
+    }
+
+    if(update_avatar) {
+      if(profile == null || profile.profile_image_url_https == null) {
+        log.warn("Twitter", "Tried to preform a user avatar update with a null profile.");
+        return;
+      }
+      util.uploadContentFromUrl(this._bridge, profile.profile_image_url_https, intent).then((uri) =>{
+        return intent.setAvatarUrl(uri);
+      }).catch(err => {
+        log.error(
+            'Twitter',
+            "Couldn't set new avatar for @%s because of %s",
+            profile.screen_name,
+            err
+          );
+      });
+    }
+    this.storage.cache_user_profile(profile.id, profile.screen_name, profile, ts);
+  });
+}
+
+MatrixTwitter.prototype._create_twitter_client = function (creds) {
+  var ts = new Date().getTime();
+  var client = new Twitter({
+    consumer_key: this.app_auth.consumer_key,
+    consumer_secret: this.app_auth.consumer_secret,
+    access_token_key: creds.access_token,
+    access_token_secret: creds.access_token_secret
+  });
+  client.last_auth = ts;
```

Comment:
Adding random properties onto objects you don't control is risky, you don't know if you're going to clobber something that was there. In this case it's probably fine, but you should **definitely** comment why you're doing this.